docs/docs/routing.md

```diff
@@ -0,0 +1,238 @@
+      this.props.onClick(event);
+    }
+
+    // Don't navigate if this.props.onClick called
+    // e.preventDefault()
+    if (event.defaultPrevented) {
+      return;
+    }
+
+    // If target prop is set (e.g. to "_blank"),
+    // let browser handle link.
+    if (this.props.target) {
+      return;
+    }
+    
+    const isNotLeftClick = event.button !== 0;
+    const isModified = !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
+    if (isModified || isNotLeftClick) {
+      return;
+    }
+
+    // Ensure the browser doesn't navigate
+    event.preventDefault();
+    
+    // browserHistory.push() expects an object that takes the
+    // same shape as `window.location`
+    const pathname = this.props.href.split('?')[0];
+    const search = this.props.href.slice(pathname.length);
+    const location = { pathname, search };
+    
+    // Update the browser URL and notify any listeners
+    // added via browserHistory.listen()
+    browserHistory.push(location);
+  }
+
+  render() {
+    return (
+      <a {...this.props} onClick={this.handleClick}>
+        {this.props.children}
+      </a>
+    );
+  }
+}
+```
+
+Use `<Link>` wherever you would normally use `<a>`. For example, the `Menu` component from the first example would use `<Link>`:
+
+```js
+function Menu() {
+  return (
+    <div>
+      <Link href="/news">News</Link>
+      <Link href="/profile">Profile</Link>
+    </div>
+  );
+}
+```
+
+## Updating rendered content
+
+When the URL is updated by `browserHistory.push()`, by default nothing will happen. In order to update the displayed content, you will need to listen for history events and take appropriate action.
+
+Since the displayed content is decided by your `App` component, the current location will need to be available within that component's `render()` method. You could accomplish this by passing it in via `props`, but we'll take take the approach of storing it in [component state](/react/docs/state-and-lifecycle.html).
+
+Assuming we're keeping the current location under the `location` key of `this.state`, this will involve four steps:
+
+1. Setting the initial state to the current value of `browserHistory.location`.
+2. Subscribing to history events once the component has initially rendered.
+3. Updating the state whenever a new location is emitted.
+4. Unsubscribing from history events when the component is unmounted.
+
+```js
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    // Set the initial location
+    this.state = {
+      location: browserHistory.location
+    };
+  }
+  
+  componentDidMount() {
+    // Subscribe after the component has mounted, in case the
+    // component is being rendered server-side and subscribing
+    // doesn't make sense
+    this.unsubscribe = browserHistory.listen(location => {
+      // Update the component state whenever the browser's
+      // location is updated
+      this.setState({ location })
```

Comment:
Minor nit: semicolons and `({location})`, we use this style elsewhere in the docs.